nomad/job_endpoint.go

```diff
@@ -1890,6 +1890,44 @@ func (j *Job) Dispatch(args *structs.JobDispatchRequest, reply *structs.JobDispa
 		return err
 	}
 
+	// Ensure that we have only one dispatched version of this job running concurrently
+	// by comparing the idempotency token against any non-terminal versions.
```

Comment:
I think this comment could use tweaking now - you can highlight the intent without descriping the implementation (which no longer special cases non-terminals). Maybe something like:

```suggestion
	// Avoid creating new dispatched jobs for retry requests, by using the idempotency token
```